### PR TITLE
fix(prerender): only extract relative links

### PR DESCRIPTION
**Pull request by Atinux**
### 🔗 Linked issue

Related to https://github.com/nuxt/nuxt3-docs/issues/21

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The crawler should only extract the relative link. I don't know where to add a test case, I did test in the nuxt 3 upcoming docs and fixes the isssue.

### 📝 Checklist

- [x] I have linked an issue or discussion.

### CODE REVIEW

File: src/prerender.ts [hunk 1]
```diff
@@ -80,7 +80,9 @@ const LINK_REGEX = /href=['"]?([^'" >]+)/g
 function extractLinks (html: string, _url: string) {
   const links: string[] = []
   for (const match of html.matchAll(LINK_REGEX)) {
-    links.push(parseURL(match[1]).pathname)
+    const url = parseURL(match[1])
+    // Extract only relative links
```
Comment (pi0) [inline src/prerender.ts:84]: Currently we don't support relative links either (see `_url` above is unused)

Comment (Atinux) [reply src/prerender.ts:84]: What I meant is excluding the url that includes a `host` (external urls).

Comment (Atinux) [reply src/prerender.ts]: Updated the comment

**Comment by pi0** [comment]
Added tests and an improvement to pass relative links as well.

**Pull request merged**